app/assets/src/components/visualizations/TidyTree.js

```diff
@@ -171,6 +179,31 @@ export default class TidyTree {
     });
   }
 
+  persistCollapsedInUrl(node) {
+    try {
```

Comment:
Here is where you want to use callbacks. This code should be on the visualization file `TaxonTreeViz`.
Here `TidyTree should only call `onCollapsedStateChange` for instance.